fip-0015.md

```diff
@@ -0,0 +1,70 @@
+---
+fip: 15
+title: CLIO Enhancements
+status: Draft
+type: Enhancements
+author: Adam Androulidakis
+created: 8/27/2020
+updated: 8/27/2020
+---
+
+# Abstract
+This proposal addresses enhancements to the FIO Protocol command line utility known as CLIO:
+* Create additional subcommands to support FIO actions
+* Cleanup and repair of existing subcommands
+* Handling of additional FIO error codes
+
+# Motivation
+The command line utility has not been maintained and deserves some customization for FIO before handing off to our community of block producers.
+
+# Specification
+## Added new subcommands
+|Option|Subcommand|Description|
+|---|---|---|
+|Request|new|Create a new FIO Request|
+|Request|reject|Reject a FIO Request|
+|Request|cancel|Cancel a FIO Request|
+|Data|record|Record an on-blockchain transaction|
+|Domain|burn_expired|Process expired domains|
+|Domain|register|Register a new FIO Domain|
+|Domain|renew|Renew a FIO Domain|
+|Domain|set_public|Set a FIO Domain to Public or Private|
+|Domain|transfer|Transfer a FIO Domain to a new owner|
+|Address|register|Register a FIO Address|
+|Address|add_pub|Add a Public Blockchain Address to a FIO Address|
+|Address|remove_pub|Remove Public Blockchain Addresses from a FIO Address|
+|Address|renew|Renew a FIO Address|
+|Address|remove_all_pub|Remove all Public Blockchain Addresses from a FIO Address|
+|Address|transfer|Transfer a FIO Address to a new owner|
+|Fee|set_amount|Set endpoint fees (for producers)|
+|Fee|set_multiplier|Set the fee multiplier|
+|System|claimrewards|Claim Block Producer Rewards|
+|System|tpidclaim|Pay our rewards to TPIDs|
+|Convert|fiokey_to_account||
+|Transfer||Transfer FIO coins to a FIO public key|
+
+## Removed subcommands
+|Option|Subcommand|
+|---|---|
+|system|buyram|
+|system|sellram|
+|currency|retire|
+|currency|transfer|
+
+## Added functionality
+|Feature|Description|
+|---|---|
+|Rename keosd|keosd folder, binaries and symbols renamed fio-wallet to avoid confict with keosd used on systems with a shared EOS installation|
+|Default max_fee|max_fee is set to 800000000001LL so user does not have to define manually. Any required fee set by the producers under this value will process|
+|Custom Push Endpoints|Certain transactions are pushed to a FIO chain api push endpoint depending on the action. Any undefined go to /v1/chain/push_transaction|
```

Comment:
800000000001LL ?

Reply:
It was just the largest fee configured at the time + 1 SUF. Do you have a recommendation ?

Reply:
Corrected, removed LL